src/main/scala/state/FunctionPreconditionTransformer.scala

```diff
@@ -40,7 +40,8 @@ object FunctionPreconditionTransformer {
           Quantification(Forall, vars, tBody, triggers, name, isGlobal, weight)
         }
       case App(hdf@HeapDepFun(_, _, _), args)  =>
-          And(args.map(transform(_, p)) :+ App(functionSupporter.preconditionVersion(hdf), args))
+        // TODO jga ???
+          And(args.map(transform(_, p)) :+ App(functionSupporter.limitedVersion(hdf), args))
```

Comment:
Just checking: Since you got rid of the precondition version of functions, you don't need the FunctionPreconditionTransformer at all any more, right?